common/scala/src/whisk/common/ConsulClient.scala

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2015-2016 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.common
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.language.postfixOps
+
+import org.apache.commons.codec.binary.Base64
+
+import akka.actor.ActorSystem
+import spray.client.pipelining.Delete
+import spray.client.pipelining.Get
+import spray.client.pipelining.Put
+import spray.client.pipelining.WithTransformerConcatenation
+import spray.client.pipelining.sendReceive
+import spray.client.pipelining.sendReceive$default$3
+import spray.client.pipelining.unmarshal
+import spray.http.HttpRequest
+import spray.http.HttpResponse
+import spray.http.Uri
+import spray.http.Uri.Path
+import spray.http.Uri.Query
+import spray.httpx.SprayJsonSupport.sprayJsonUnmarshaller
+import spray.json.DefaultJsonProtocol
+
+case class ConsulEntry(Key: String, Value: String) {
```

Comment:
Ah makes sense. I would still favor writing a custom serializer over using non-conventional Scala (and add the comment in the serializer). `Value` in the code looks like a constant/object.